### ISSUE TITLE
Could not load @nkohari/apocrypha/catalog

### ISSUE THREAD

**Issue by antonio-ivanovski** [nkohari/apocrypha#1]
When I try to start/build the https://github.com/nkohari/nate.io project with `npm build` I get this error:

```
[vite-plugin-apocrypha] Could not load @nkohari/apocrypha/catalog (imported by src/App.tsx): Cannot read properties of undefined (reading 'id')
error during build:
TypeError: Cannot read properties of undefined (reading 'id')
    at /Users/antonioivanovski/dev/nate.io/vite.config.ts:257:15
    at /Users/antonioivanovski/dev/nate.io/vite.config.ts:247:14
    at /Users/antonioivanovski/dev/nate.io/vite.config.ts:304:12
    at /Users/antonioivanovski/dev/nate.io/vite.config.ts:247:14
    at SpotifyClient.request (/Users/antonioivanovski/dev/nate.io/vite.config.ts:396:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async DiskCache.readThrough (/Users/antonioivanovski/dev/nate.io/vite.config.ts:189:21)
    at async /Users/antonioivanovski/dev/nate.io/vite.config.ts:426:19
    at async DocumentFactory.getMetadata (/Users/antonioivanovski/dev/nate.io/node_modules/@nkohari/apocrypha/dist/index.js:16867:22)
    at async DocumentFactory.create (/Users/antonioivanovski/dev/nate.io/node_modules/@nkohari/apocrypha/dist/index.js:16844:22)
```

Full log: https://gist.github.com/toteto/d46cd512c17fd25be4b5ab6f720380e0

Before running the script, I ran the `npm install`. I see no further instructions on how to get this tool working.

**Comment by nkohari** [comment]
You're missing an `.env` file which contains Spotify authentication information. The `nate.io` project isn't really meant to be built by anyone other than me. I released the source in the hopes someone would find it useful, but it's really just my personal website.

If you really do want to get it running, it'll require some hackery. You'll first need to create a Spotify app, and then add an `.env` file with the following information. (The refresh token is kind of a pain to get; I manually walked through an OAuth flow to get one.)

```
SPOTIFY_CLIENT_ID=<your spotify client id>
SPOTIFY_CLIENT_SECRET=<your spotify client secret>
SPOTIFY_REFRESH_TOKEN=<a spotify refresh token received from an oauth flow>
```

You don't need to build the `nate.io` project to use Apocrypha, though!
